packages/integration-sdk-runtime/src/synchronization/index.ts

```diff
@@ -524,42 +550,92 @@ function getLargestRawDataEntryFromEntity(
 }
 
 /**
- * Removes data from the rawData of the largest entity until the overall size
- * of the data object is less than maxSize (defaulted to UPLOAD_SIZE_MAX).
+ * Helper function to generate a map of property sizes for a given Entity.
+ * This is used to determine what property keys on an Entity have a particularly large value.
+ * Intended to be used when logging upload errors due to large payload size.
  *
  * @param data
+ * @returns
  */
-export function shrinkRawData(
-  data: UploadDataLookup[keyof UploadDataLookup][],
-  maxSize = UPLOAD_SIZE_MAX,
+function getPropSizeMapFromEntity(data: Entity): any {
+  const propSizeMap = {};
+
+  for (const [key, value] of Object.entries(data)) {
+    propSizeMap[key] = Buffer.byteLength(JSON.stringify(value));
+  }
+
+  return propSizeMap;
+}
+
+/**
+ * Removes data from the rawData of the largest entity until the overall size
+ * of the data object is less than maxSize (defaulted to MAX_BATCH_SIZE).
+ *
+ * @param batchData
+ */
+export function shrinkBatchRawData(
+  batchData: UploadDataLookup[keyof UploadDataLookup][],
+  logger: IntegrationLogger,
+  maxBatchSize = MAX_BATCH_SIZE,
 ): ShrinkRawDataResults {
   const startTimeInMilliseconds = Date.now();
-  let totalSize = Buffer.byteLength(JSON.stringify(data));
-  const initialSize = totalSize;
+  let totalBatchSize = Buffer.byteLength(JSON.stringify(batchData));
+  const initialBatchSize = totalBatchSize;
   let itemsRemoved = 0;
   const sizeOfTruncated = Buffer.byteLength("'TRUNCATED'");
 
-  while (totalSize > maxSize) {
-    // Find largest Entity
-    const largestEntity = getLargestEntityFromBatch(data);
+  while (totalBatchSize > maxBatchSize) {
+    // Find Entity with largest rawData
+    const entityWithLargestRawData =
+      getEntityFromBatchWithLargestRawData(batchData);
 
     // If we don't have any entities to shrink or the optional _rawData array is empty,
     // we have no other options than to throw an error.
-    if (largestEntity?._rawData) {
-      // Find largest _rawData entry (typically 0, but check to be certain)
-      const largestRawDataEntry = getLargestRawDataEntryFromEntity(
-        largestEntity._rawData,
+    if (entityWithLargestRawData?._rawData) {
+      // Find largest element of the _rawData array (typically at index 0, but check to be certain)
+      const largestRawDataEntry = getLargestRawDataEntry(
+        entityWithLargestRawData._rawData,
       );
-
-      // Find largest item within rawData
+      // Find largest item within largest that element
```

Comment:
nit:  this comment doesn't quite read the way I think you meant it to

Reply:
no it does not :)